manager/controlapi/task.go

```diff
@@ -104,10 +104,10 @@ func (s *Server) ListTasks(ctx context.Context, request *api.ListTasksRequest) (
 	if request.Filters != nil {
 		tasks = filterTasks(tasks,
 			func(e *api.Task) bool {
-				return filterContains(e.ServiceAnnotations.Name, request.Filters.Names)
+				return filterContains(e.Annotations.Name, request.Filters.Names)
```

Comment:
This may need to handle falling back to old task names when `e.Annotations.Name` is unset.